lib/Doctrine/Common/Annotations/CachedReader.php

```diff
@@ -265,4 +267,9 @@ private function getTraitLastModificationTime(ReflectionClass $reflectionTrait):
 
         return $this->loadedFilemtimes[$fileName] = $lastModificationTime;
     }
+
+    private static function getFileMtime(string $fileName): int
+    {
+        return file_exists($fileName) && is_file($fileName) ? filemtime($fileName) : 0;
```

Comment:
Why do we need to call both `file_exists` and `is_file`?

Reply:
``` php
return file_exists($fileName) ? filemtime($fileName) : 0;
```
should do the trick too, shouldn't it?

Reply:
The edge case that ` is_file`  checks for here is if ` $fileName` is a directory. I guess it makes sense to return 0 in this case, but I don't know if :
1. It's the focus of this PR
2.  `getFileMtime()` can get called with a directory name at all

Reply:
Isn't `file_exists` redundant if we have `is_file`?

Reply:
[indeed](https://www.php.net/manual/en/function.is-file.php#refsect1-function.is-file-returnvalues).